src/main/java/frc/robot/vision/aprilTags/AprilTagFilters.java

```diff
@@ -0,0 +1,65 @@
+package frc.robot.vision.aprilTags;
+
+import edu.wpi.first.math.geometry.*;
+import frc.robot.vision.VisionRawData;
+
+public class AprilTagFilters {
+
+	public static boolean keepRobotPoseData(VisionRawData visionData, Pose2d currentEstimatedPose, AprilTagFiltersTolerances tolerances) {
+		return AprilTagFilters.isLimelightOutputInTolerance(
+			visionData,
+			currentEstimatedPose,
+			tolerances.normalizedPositionTolerance(),
+			tolerances.normalizedRotationTolerance()
+		)
+			&& AprilTagFilters.isRollInTolerance(visionData, tolerances.rollTolerance())
+			&& AprilTagFilters.isPitchInTolerance(visionData, tolerances.pitchTolerance())
+			&& AprilTagFilters.isRobotOnGround(visionData, tolerances.robotToGroundToleranceMeters())
+			&& !AprilTagFilters.isDataTooAmbiguous(visionData, tolerances.maximumAmbiguity())
+			&& !AprilTagFilters.isLatencyTooHigh(visionData, tolerances.maximumLatency());
+	}
```

Comment:
again, I think there should be separate classes for the filters of the photon and limelight. There could be filters that would only fit photon and will not fit limelight. Also, we can't get the ambiguity of the limelights